airflow/contrib/operators/gcp_transfer_operator.py

```diff
@@ -0,0 +1,856 @@
+
+    :param job_name: Name of the job to be updated
+    :type job_name: str
+    :param body: The request body, as described in
+        https://cloud.google.com/storage-transfer/docs/reference/rest/v1/transferJobs/patch#request-body
+        With three additional improvements:
+
+        * dates can be given in the form datetime.date
+        * times can be given in the form datetime.time
+        * credentials to Amazon Web Service should be stored in connection and
+          indicated by the aws_conn_id parameter
+
+    :type body: dict
+    :param aws_conn_id: The connection ID used to retrieve credentials to
+        Amazon Web Service.
+    :type aws_conn_id: str
+    :param gcp_conn_id: The connection ID used to connect to Google Cloud
+        Platform.
+    :type gcp_conn_id: str
+    :param api_version: API version used (e.g. v1).
+    :type api_version: str
+    :return: Google Storage Transfer service object
+    :rtype: dict
+    """
+    # [START gcp_transfer_job_update_template_fields]
+    template_fields = ('job_name', 'body', 'gcp_conn_id', 'aws_conn_id', )
+    # [END gcp_transfer_job_update_template_fields]
+
+    @apply_defaults
+    def __init__(self,
+                 job_name,
+                 body,
+                 aws_conn_id='aws_default',
+                 gcp_conn_id='google_cloud_default',
+                 api_version='v1',
+                 *args,
+                 **kwargs):
+        super(GcpTransferServiceJobUpdateOperator, self).__init__(*args, **kwargs)
+        self.job_name = job_name
+        self.body = body
+        self.gcp_conn_id = gcp_conn_id
+        self.api_version = api_version
+        self.aws_conn_id = aws_conn_id
+        self._validate_inputs()
+
+    def _validate_inputs(self):
+        TransferJobValidator(
+            body=self.body,
+        ).validate_body()
+        if not self.job_name:
+            raise AirflowException("The required parameter 'job_name' "
+                                   "is empty or None")
+
+    def execute(self, context):
+        TransferJobPreprocessor(
+            body=self.body,
+            aws_conn_id=self.aws_conn_id,
+        ).process_body()
+        hook = GCPTransferServiceHook(
+            api_version=self.api_version,
+            gcp_conn_id=self.gcp_conn_id,
+        )
+        return hook.update_transfer_job(
+            job_name=self.job_name,
+            body=self.body,
+        )
+
+
+class GcpTransferServiceJobDeleteOperator(BaseOperator):
+    """
+    Delete a transfer job.
+
+    .. seealso::
+        For more information on how to use this operator, take a look at the guide:
+        :ref:`howto/operator:GcpTransferServiceJobDeleteOperator`
+
+    :param job_name: The name of the operation resource to be cancelled.
+    :type job_name: str Name of the transfer job. Required.
+    :type gcp_conn_id: str
+    :param project_id: Optional, Google Cloud Platform Project ID.
```

Comment:
The optionality here is inconsistent: here it is `: Optional, ...` (at the beginning) and in l. 289 it's `... Required.` (at the end).
It would be good to stick to a consistent standard with regards to this, at least in a single PR.

You might take a look at the Cloud Vision operators and how I did it there: https://github.com/apache/airflow/blob/cbc8a2c665f865222318273acab01c759d33b1a5/airflow/contrib/operators/gcp_vision_operator.py

Reply:
I updated a description of operators and method in hooks.